Source/Dafny/DafnyOptions.cs

```diff
@@ -980,19 +933,89 @@ not sound
+    go - Compilation to Go
+    js - Compilation to JavaScript
+    java - Compilation to Java
+    cpp - Compilation to C++
+    php - Compilation to PHP
+
+    Note that the C++ backend has various limitations
+    (see Docs/Compilation/Cpp.md). This includes lack of support for
+    BigIntegers (aka int), most higher order functions, and advanced
+    features like traits or co-inductive types.
+/Main:<name>
+    The (fully-qualified) name of the method to use as the executable entry point.
+    Default is the method with the {{:main}} atrribute, or else the method named 'Main'.
+/compileVerbose:<n>
+    0 - don't print status of compilation to the console
+    1 (default) - print information such as files being written by
+        the compiler to the console
+/spillTargetCode:<n>
+    This option concerns the textual representation of the target program.
+    This representation is of no interest when working with only Dafny code,
+    but may be of interest in cross-language situations.
+    0 (default) - Don't make any extra effort to write the textual target
+        program (but still compile it, if /compile indicates to do so).
+    1 - Write the textual target program, if it is being compiled.
+    2 - Write the textual target program, provided it passes the verifier (and
+        /noVerify is NOT used), regardless of /compile setting.
+    3 - Write the textual target program, regardless of verification outcome
+        and /compile setting.
+    Note, some compiler targets may (always or in some situations) write out
+    the textual target program as part of compilation, in which case
+    /spillTargetCode:0 behaves the same way as /spillTargetCode:1.
+/out:<file>
+    filename and location for the generated target language files
+/coverage:<file>
+    The compiler emits branch-coverage calls and outputs into
+    <file> a legend that gives a description of each
+    source-location identifier used in the branch-coverage calls.
+    (use - as <file> to print to console)
+/optimize     Produce optimized C# code, meaning:
+      - passes /optimize flag to csc.exe.
+/optimizeResolution:<n>
+    0 - Resolve and translate all methods
+    1 - Translate methods only in the call graph of current verification target
+    2 (default) - As in 1, but only resolve method bodies in non-included Dafny
+                  sources
+/useRuntimeLib
+    Refer to pre-built DafnyRuntime.dll in compiled assembly rather
+    than including DafnyRuntime.cs verbatim.
+
+  ---- Dafny test generation options -----------------------------------------
+
+{TestGenOptions.Help}
+
+  ---- Pass-through Boogie options -------------------------------------------
 
 Dafny generally accepts Boogie options and passes these on to Boogie. However,
 some Boogie options, like /loopUnroll, may not be sound for Dafny or may not
 have the same meaning for a Dafny program as it would for a similar Boogie
-program.";
+program." + base.Help;
```

Comment:
This won't work: base.Help contains the command line header that we want to show first.  You really want base.base.help + Dafny + Boogie